src/vuestic-theme/vuestic-components/vuestic-checkbox/VuesticCheckbox.vue

```diff
@@ -1,20 +1,42 @@
 <template>
-  <div class="vuestic-checkbox form-check abc-checkbox"
-       :class="additionalClasses"
+  <div
+    class="vuestic-checkbox"
+    :class="{'vuestic-checkbox--selected': selected, 'vuestic-checkbox--readonly': readonly,
+    'vuestic-checkbox--disabled': disabled, 'vuestic-checkbox--error': errorComputed, 'vuestic-checkbox--onfocus': isOnFocus}"
   >
-    <input class="form-check-input"
-           type="checkbox"
-           :id="id"
-           :name="name"
-           :checked="checked"
-           @change="onChange"
-           :disabled="disabled"
-    >
-    <label class="form-check-label" :for="id">
-      <span class="abc-label-text">
-        <slot>{{ label }}</slot>
+    <div class="vuestic-checkbox__content">
+      <div class="vuestic-checkbox__square" @click="toggleSelection">
+        <div class="vuestic-checkbox__input-container">
+          <input
+            :id="id"
+            :readonly="true"
+            @focus="isOnFocus = true"
+            @blur="isOnFocus = false"
+            class="vuestic-checkbox__input"
+            @keypress="toggleSelection"
+          />
+        </div>
+        <span class="vuestic-checkbox__icon-selected-container">
+          <i class="ion ion-md-checkmark vuestic-checkbox__icon-selected" aria-hidden="true"/>
       </span>
-    </label>
+      </div>
+      <div class="vuestic-checkbox__label-container">
+        <span
```

Comment:
Why wouldn't we use `label` tag here?

Reply:
@jean-moldovan that way we don't need to gen id.

Reply:
ok